pyscf: init at 1.7.6

sheepforce opened the pull request:
###### Motivation for this change
Adds pyscf, a quantum chemistry software written in python. Part of the packages @markuskowa and me want to merge from markuskowa/NixOS-QChem#56 .

The test suite is disabled, as it is for some reasons very sensitive to the filesystem, the build is running on. @markuskowa and my journey with this here: https://github.com/markuskowa/NixOS-QChem/issues/29

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

markuskowa commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,43 @@
+
+  # The test suite is very sensitive to the file system and built only correctly on tmpfs.
+  # Therefore disabled.
+  doCheck = false;
```
Can ofBorg build it when you turn on the tests?

sheepforce replied:
I will just push a commit to check this.

markuskowa replied:
Seems to work. I suggest keeping it turned on and remove the comment. We never double checked the tmpfs hypothesis.

sheepforce replied:
OK, turned them on and removed comment!

r-rmcgibbo commented:
Result of `nixpkgs-review pr 127880` at b58bdd8b run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/91a81ef2fbd2ce38e64a2f6d62c62698">python38Packages.pyscf</a></li>
    <li><a href="https://gist.github.com/f98606bfcc604bb9c3039864d39df940">python39Packages.pyscf</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/development/python-modules/pyscf/default.nix:38:16:
```
   |
38 |     homepage = https://pyscf.github.io/;
   |                ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

markuskowa commented:
@GrahamcOfBorg build python3Packages.pyscf

markuskowa commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,43 @@
+
+  meta = with lib; {
+    description = "Python-based simulations of chemistry framework";
+    homepage = https://pyscf.github.io/;
```
```suggestion
    homepage = "https://pyscf.github.io/";
```

sheepforce committed:
pyscf: init at 1.7.6.post1


pyscf: adding comment regarding tests


pyscf: enable test for test with ofBorg


pyscf: quote url


pyscf: keep tests turned on and remove comment

sheepforce committed:
pyscf: formatting and removing python3 attribute


pyscf: adding comment regarding tests


pyscf: enable test for test with ofBorg


pyscf: quote url


pyscf: keep tests turned on and remove comment

SuperSandro2000 commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  meta = with lib; {
+    description = "Python-based simulations of chemistry framework";
+    homepage = "https://pyscf.github.io/";
```
```suggestion
    homepage = "https://github.com/pyscf/pyscf";
```

SuperSandro2000 commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  PYSCF_INC_DIR="${libcint}:${libxc}:${xcfun}";
+
+  doCheck = true;
```
checkInputs and pythonImportsCheck missing.

sheepforce replied:
checkInputs is probably empty. I can't run any of the tests on my machine unfortunately. As mentioned above I have probably some file system issues. But at least it doesn't use pytest or anything as far as I can see.

sheepforce replied:
Disabled the tests again and therefore removed `checkInputs` as suggested by @markuskowa below. Keeping `pythonImportsCheck`.

SuperSandro2000 commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,41 @@
+    numpy
+    scipy
+    h5py
+    python3
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,41 @@
+{ buildPythonPackage, lib, fetchFromGitHub, libcint, libxc, xcfun, blas,
+  numpy, scipy, h5py, python3
+} :
```
```suggestion
{ buildPythonPackage, lib, fetchFromGitHub, libcint, libxc, xcfun, blas
, numpy, scipy, h5py
} :
```

SuperSandro2000 commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+buildPythonPackage rec {
+  pname = "pyscf";
+  version = "1.7.6.post1";
```
```suggestion
  version = "1.7.6";
```

markuskowa replied:
@sheepforce Was there a particular reason for using a "non-release" commit?

sheepforce replied:
This is not exactly 1.7.6 but actually the post release as labeled on PyPi. To be consistent I think it might be better to keep the version that is also on PyPi?

sheepforce replied:
XCFun does not work with 1.7.6. 1.7.6.post1 addresses this and only this. This is basically the release version as on PyPi. https://pypi.org/project/pyscf/1.7.6.post1/
It just does not have a git tag.

SuperSandro2000 commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,41 @@
+  src = fetchFromGitHub {
+    owner = "pyscf";
+    repo = pname;
+    rev = "f6c9c6654dd9609c5e467a1edd5c2c076f793acc";
```
```suggestion
    rev = "v${version}";
```

sheepforce replied:
See above. 1.7.6 does not build correctly

sheepforce replied:
https://github.com/pyscf/pyscf/issues/872

markuskowa replied:
Sounds like it boils down to a single issue. Is it also a single (or a few) commits that fix the issue? If yes, you could simply pull them in as patches.

sheepforce replied:
It is just 2-3 commits behind the release tag. Pulling these two as patches would be exactly equivalent to this commit. So I guess the current version is easier?

SuperSandro2000 replied:
Using version makes automated updates easier and I do not need to verify that not a commit from a fork is fetched.

sheepforce replied:
@SuperSandro2000 So I should use the release tag and fetch the missing commits as patches as @markuskowa suggested? The 1.7.6 release tag alone will not work.

sheepforce replied:
And as written above 1.7.6.post1 is exactly the version tag that is on PyPi.

sheepforce committed:
pyscf: change homepage to github repo

sheepforce committed:
pyscf: adding checkInputs and pythonImportsCheck

markuskowa commented:
One the changes to testing framework broke the build.

sheepforce commented:
> One the changes to testing framework broke the build.

Seems to be the empty `checkInputs`. At least the `pythonImportChecks` work for me. Unfortunately I cannot narrow it down easily. I can't find any machine in the office where the test ever run correctly, while ofBorg apparently built them correctly :grimacing:

markuskowa commented:
Well, I give up: ofBorg fails at the test suite but locally it passes. Let's just disable the tests. The test suite is too unreliable. It was worth a try.

sheepforce committed:
pyscf: disable tests

markuskowa reviewed (approved):
I am OK with having "1.7.6.post1" and fixed rev hash. This prevents auto updating, but using patches instead would have the same problem.

SuperSandro2000 commented on pkgs/development/python-modules/pyscf/default.nix:
```diff
@@ -0,0 +1,41 @@
+{ buildPythonPackage, lib, fetchFromGitHub, libcint, libxc, xcfun, blas
+, numpy, scipy, h5py
+} :
```
```suggestion
}:
```

SuperSandro2000 committed:
Update pkgs/development/python-modules/pyscf/default.nix

drewrisinger commented:
I'm late to the party, but I separately made a derivation for pyscf (in NUR) a year ago, which runs the tests. Here it is if you're interested. Now that this is in-tree I'm considering dropping my derivation.
https://github.com/drewrisinger/nur-packages/blob/6f872a9bbd083f0f5d953eb8403c41df318f482e/pkgs/python-modules/pyscf/default.nix

sheepforce commented:
Very nice :slightly_smiling_face: Running the test suite (or at least parts of it) is for sure a good idea. If you find some time, it would be great if you could maybe add the necessary stuff for tests. If not I can also try to do this later, when I find some time :slightly_smiling_face:
